src/Simplex/Chat/Migrations/M20230914_member_probes.hs

```diff
@@ -8,25 +8,164 @@ import Database.SQLite.Simple.QQ (sql)
   [sql|
-ALTER TABLE group_members ADD COLUMN sent_probe BLOB;
-ALTER TABLE group_members ADD COLUMN received_probe BLOB;
-ALTER TABLE group_members ADD COLUMN received_probe_hash BLOB;
+CREATE TABLE new__sent_probes(
+  sent_probe_id INTEGER PRIMARY KEY,
+  contact_id INTEGER REFERENCES contacts ON DELETE CASCADE,
+  group_member_id INTEGER REFERENCES group_members ON DELETE CASCADE,
+  probe BLOB NOT NULL,
+  user_id INTEGER NOT NULL REFERENCES users ON DELETE CASCADE,
+  created_at TEXT CHECK(created_at NOT NULL),
+  updated_at TEXT CHECK(updated_at NOT NULL),
+  UNIQUE(user_id, probe)
+);
 
-CREATE INDEX idx_group_members_sent_probe ON group_members(sent_probe);
-CREATE INDEX idx_group_members_received_probe_hash ON group_members(received_probe_hash);
+CREATE TABLE new__sent_probe_hashes(
+  sent_probe_hash_id INTEGER PRIMARY KEY,
+  sent_probe_id INTEGER NOT NULL REFERENCES new__sent_probes ON DELETE CASCADE,
+  contact_id INTEGER REFERENCES contacts ON DELETE CASCADE,
+  group_member_id INTEGER REFERENCES group_members ON DELETE CASCADE,
+  user_id INTEGER NOT NULL REFERENCES users ON DELETE CASCADE,
+  created_at TEXT CHECK(created_at NOT NULL),
+  updated_at TEXT CHECK(updated_at NOT NULL)
+);
+
+CREATE TABLE new__received_probes(
+  received_probe_id INTEGER PRIMARY KEY,
+  contact_id INTEGER REFERENCES contacts ON DELETE CASCADE,
+  group_member_id INTEGER REFERENCES group_members ON DELETE CASCADE,
+  probe BLOB,
+  probe_hash BLOB NOT NULL,
+  user_id INTEGER NOT NULL REFERENCES users ON DELETE CASCADE,
+  created_at TEXT CHECK(created_at NOT NULL),
+  updated_at TEXT CHECK(updated_at NOT NULL)
+);
+
+INSERT INTO new__sent_probes
+  (sent_probe_id, contact_id, probe, user_id, created_at, updated_at)
+SELECT
+  sent_probe_id, contact_id, probe, user_id, created_at, updated_at
+  FROM sent_probes;
+
+INSERT INTO new__sent_probe_hashes
+  (sent_probe_hash_id, sent_probe_id, contact_id, user_id, created_at, updated_at)
+SELECT
+  sent_probe_hash_id, sent_probe_id, contact_id, user_id, created_at, updated_at
+  FROM sent_probe_hashes;
+
+INSERT INTO new__received_probes
+  (received_probe_id, contact_id, probe, probe_hash, user_id, created_at, updated_at)
+SELECT
+  received_probe_id, contact_id, probe, probe_hash, user_id, created_at, updated_at
+  FROM received_probes;
+
+DROP INDEX idx_sent_probe_hashes_user_id;
+DROP INDEX idx_sent_probe_hashes_contact_id;
+DROP INDEX idx_received_probes_user_id;
+DROP INDEX idx_received_probes_contact_id;
+
+DROP TABLE sent_probes;
+DROP TABLE sent_probe_hashes;
+DROP TABLE received_probes;
+
+ALTER TABLE new__sent_probes RENAME TO sent_probes;
+ALTER TABLE new__sent_probe_hashes RENAME TO sent_probe_hashes;
+ALTER TABLE new__received_probes RENAME TO received_probes;
+
+CREATE INDEX idx_sent_probes_user_id ON sent_probes(user_id);
+CREATE INDEX idx_sent_probes_contact_id ON sent_probes(contact_id);
+CREATE INDEX idx_sent_probes_group_member_id ON sent_probes(group_member_id);
```

Comment:
not sure both idx_sent_probes_probe and UNIQUE(user_id, probe) are needed, probably queries would be written for the second one